WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersPeriodViewController.swift

```diff
@@ -41,16 +41,15 @@ final class StoreStatsAndTopPerformersPeriodViewController: UIViewController {
 
     /// Minimal time interval for data refresh
     var minimalIntervalBetweenSync: TimeInterval {
-        switch timeRange {
-        case .today:
+        switch timeRange.intervalGranularity {
+        case .hourly:
             return 60
-        case .thisWeek, .thisMonth:
+        case .daily, .weekly:
             return 60*60
-        case .thisYear:
+        case .monthly, .quarterly:
             return 60*60*12
-        case .custom:
-            // TODO: 11935 Specify refresh interval
-            return 60
+        case .yearly:
+            return 60*60*24*15
```

Comment:
Out of curiosity, was this formula for yearly discussed somewhere? I think it makes sense but just want to know the rationale.

Reply:
Good question! I made the decision based on the patterns from the previous calculations:
- When the time range is `Today`, the interval granularity is hourly, then the refresh rate is 60 seconds. This means the stats can be refreshed every minute to check for any new data for the next hour.
- Similarly, when the time range is `This Week` and `This Month`, the granularity is daily and weekly respectively, then the data is refreshed every 1 hour to check if it's the new day/week/month yet.
- For monthly and quarterly granularity, we check every 12 hours (half a day) for new data. 
- For yearly granularity, we don't need to refresh as often so I picked half a month for the refresh rate. Now that I'm thinking again, that rate might be too sparse, as the app will likely be relaunched within that period. I updated the rate to be daily in c4f48d82a3f88f49f445d7d52ceec824ea94d325 - please feel free to comment if you have other suggestions.

Reply:
Thanks for the explanation. I think daily for the yearly makes sense, too, and probably also more useful for users who frequently use the app.